core/kotlinx-coroutines-core/src/main/kotlin/kotlinx/coroutines/experimental/channels/ConflatedBroadcastChannel.kt

```diff
@@ -81,25 +81,29 @@ public class ConflatedBroadcastChannel<E>() : BroadcastChannel<E> {
      * It throws the original [close][SendChannel.close] cause exception if the channel has _failed_.
      */
     @Suppress("UNCHECKED_CAST")
-    public val value: E get() {
-        _state.loop { state ->
-            when (state) {
-                is Closed -> throw state.valueException
-                is State<*> -> {
-                    if (state.value === UNDEFINED) throw IllegalStateException("No value")
-                    return state.value as E
+    override var value: E
+        get() {
+            _state.loop { state ->
+                when (state) {
+                    is Closed -> throw state.valueException
+                    is State<*> -> {
+                        if (state.value === UNDEFINED) throw IllegalStateException("No value")
+                        return state.value as E
+                    }
+                    else -> error("Invalid state $state")
                 }
-                else -> error("Invalid state $state")
             }
         }
-    }
+        set(value) {
+            sendBlocking(value)
```

Comment:
`offer` should be enough.

Reply:
`offer` would never throw any exception. But I could do `if (!offer(value)) throw IllegalStateException()`.

Or do we prefer to silently fail if the broadcast channel is closed ?

Reply:
Are you sure?

https://github.com/Kotlin/kotlinx.coroutines/blob/fc6e31e206d3a024189b701903736429423fc767/core/kotlinx-coroutines-core/src/main/kotlin/kotlinx/coroutines/experimental/channels/ConflatedBroadcastChannel.kt#L201

Reply:
Yes sorry, I was wrong. I'll use `offer`. At least I learned something :-)